Support PHP 8

ACLI would likely run fine on PHP 8 if we just relax the version constraint in composer.json, but we should verify this and then change the constraint.

Currently, trying to use PHP 8 produces:
```
Box Requirements Checker
========================
> Using PHP 8.0.0
> PHP is using the following php.ini file:
  /usr/local/etc/php/8.0/php.ini
> Checking Box requirements:
  E....
 [ERROR] Your system is not ready to run the application.
Fix the following mandatory requirements:
=========================================
 * The application requires the version "^7.3" or greater.
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -4,6 +4,7 @@ version: ~> 1.0
 php:
   - 7.3
   - 7.4
+  - 8.0.0
 
 cache:
   directories:

--- a/composer.json
+++ b/composer.json
@@ -17,7 +17,7 @@
     }
   ],
   "require": {
-    "php": "^7.3",
+    "php": "^7.3 | ^8.0",
     "ext-json": "*",
     "acquia/drupal-environment-detector": "^1.2.0",
     "composer/semver": "^3.2",

--- a/config/services.yml
+++ b/config/services.yml
@@ -54,7 +54,11 @@ services:
 
   # AcquiaCloudApi services.
   AcquiaCloudApi\Connector\Client: ~
-  AcquiaCloudApi\Connector\ConnectorInterface: ~
+  AcquiaCloudApi\Connector\Connector:
+    arguments:
+      $config: 'something'
+  AcquiaCloudApi\Connector\ConnectorInterface:
+    alias: AcquiaCloudApi\Connector\Connector
   AcquiaLogstream\LogstreamManager: ~
 
   # Symfony services.

--- a/config/services.yml
+++ b/config/services.yml
@@ -56,7 +56,7 @@ services:
   AcquiaCloudApi\Connector\Client: ~
   AcquiaCloudApi\Connector\Connector:
     arguments:
-      $config: 'something'
+      $config: ['@=service("datastore.cloud").get("key")', '@=service("datastore.cloud").get("secret")']
   AcquiaCloudApi\Connector\ConnectorInterface:
     alias: AcquiaCloudApi\Connector\Connector
   AcquiaLogstream\LogstreamManager: ~

